### PR TITLE
Hosting CLI: use http endpoint to return deploy milestones

### PR DESCRIPTION
**Pull request by martinxu9**
### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/reflex-dev/reflex/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/reflex-dev/reflex/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### New Feature Submission:

- [x] Does your submission pass the tests? 
- [x] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### **After** these steps, you're ready to open a pull request.
- For users with less ideal internet conditions, it's possible the websocket streaming deploy milestones get disconnected. There was no retry mechanism. Instead, hosting service adds an equivalent endpoint to query these events once. CLI can query this endpoint peridiocally.

### CODE REVIEW

**Comment by martinxu9** [comment]
@ElijahAhianyo could you help me try this PR for deploying an app? I think you ran into some issues https://discord.com/channels/1029853095527727165/1168071548817645599/1168090802740809778 I just switched out the websocket and replaced with http and retries. Let me know if it helps with this particular situation. There'll be a different PR for the overall slowness.

File: reflex/utils/hosting.py [hunk 1]
```diff
@@ -726,7 +728,22 @@ def get_deployment_status(key: str) -> DeploymentStatusResponse:
...
+def convert_to_local_time_with_tz(iso_timestamp: str) -> datetime | None:
+    """Helper function to convert the iso timestamp to local time.
+
+    Args:
+        iso_timestamp: The iso timestamp to convert.
+
+    Returns:
+        The converted timestamp with timezone.
+    """
+    try:
```
Comment (picklelo) [inline reflex/utils/hosting.py:740]: What causes this to raise an exception? Generally shouldn't catch bare Exceptions

File: reflex/utils/hosting.py [hunk 2]
```diff
@@ -736,7 +753,9 @@ def convert_to_local_time(iso_timestamp: str) -> str:
         The converted timestamp string.
     """
     try:
-        local_dt = datetime.fromisoformat(iso_timestamp).astimezone()
+        local_dt = convert_to_local_time_with_tz(iso_timestamp)
+        if local_dt is None:
+            return iso_timestamp
         return local_dt.strftime("%Y-%m-%d %H:%M:%S.%f %Z")
     except Exception as ex:
```
Comment (picklelo) [inline reflex/utils/hosting.py:760]: Same here, can we catch a more specific error / do we know why we need the catch?

File: reflex/utils/hosting.py [hunk 3]
```diff
@@ -1006,6 +1025,79 @@ def log_out_on_browser():
...
+                    maybe_timestamp := convert_to_local_time_with_tz(row["timestamp"])
+                ) is not None:
+                    console.debug(
+                        f"Updating from {from_iso_timestamp} to {maybe_timestamp}"
+                    )
+                    # Add a small delta so does not poll the same logs
+                    from_iso_timestamp = maybe_timestamp + timedelta(microseconds=1e5)
+                else:
+                    console.error(f"Unable to parse timestamp {row['timestamp']}")
+                    raise ValueError
```
Comment (picklelo) [inline reflex/utils/hosting.py:1081]: Should we specify an error in the ValueError? Also, do we want to break out of the loop here or just continue?

File: reflex/utils/hosting.py (hunk 1 above)
Comment (martinxu9) [reply reflex/utils/hosting.py:740]: the conversion can raise TypeError, ValueError, let me change to the more specific exception types.

File: reflex/utils/hosting.py (hunk 2 above)
Comment (martinxu9) [reply reflex/utils/hosting.py]: Actually we don't need to catch here, it's already caught in the call to convert.

File: reflex/utils/hosting.py (hunk 3 above)
Comment (martinxu9) [reply reflex/utils/hosting.py]: Good catch, we should continue here instead.

**Review by picklelo — APPROVED** [review]

**Pull request merged**